aten/src/ATen/BatchingRegistrations.cpp

```diff
@@ -233,6 +233,24 @@ Tensor unsqueeze_batching_rule(const Tensor& self, int64_t dim) {
   return self_physical.newLogicalFromPhysical(result);
 }
 
+Tensor& fill_inplace_scalar_batching_rule(Tensor& self, Scalar value) {
+  auto self_physical = MultiBatchVmapTransform::logicalToPhysical(self);
+  self_physical.tensor().fill_(value);
+  return self;
+}
+
+Tensor& fill_inplace_tensor_batching_rule(Tensor& self, const Tensor& value) {
+  auto self_physical = MultiBatchVmapTransform::logicalToPhysical(self);
+  self_physical.tensor().fill_(value);
+  return self;
+}
```

Comment:
Can you test what happens when
```
x = torch.randn(3, 5)
y = torch.randn(3)
vmap(Tensor.fill_)(x, y)
```
?
I think this will probably infinitely loop how it is written. We should either:
- add a check that `value` is not a batchedtensor (isBatchedTensor)
- handle the case where `value` is a batchedtensor and apply a copy_ instead of a fill
```

Reply:
Thank you so much for the detailed suggestions, I have updated this PR to support the `value` as a batchedtensor.

Reply:
>this will probably infinitely loop how it is written
I tried the above test code before updating the PR according to the code review.
It seems the `VmapTransforms.cpp` is able to catch this case and throw a Runtime Error.
```
Traceback (most recent call last):
  File ".\testcode\fill_vmap.py", line 13, in <module>
    vmap(torch.Tensor.fill_)(x, y)
  File "e:\repo\ml\github\pytorch\torch\_vmap_internals.py", line 250, in wrapped
    batched_outputs = func(*batched_inputs)
RuntimeError: batched INTERNAL ASSERT FAILED at "..\\aten\\src\\ATen\\VmapTransforms.cpp":45, please report a bug to PyTorch. logicalToPhysical(tensor) should only be passed a BatchedTensor
```